### PR TITLE
feat: add products component

### PR DESCRIPTION
**Pull request by jonathanzhang53**
## Description
Added product list view as stated in GCI Task by refactoring code from accounting and replicating the screen from community app.
[Task](https://codein.withgoogle.com/dashboard/task-instances/6305675132010496/)

## Related issues and discussion
Implements #328 

## GIF
Community App:
image
Web App:
ok

## Checklist

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/products/products.component.html [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  <mat-card>
+
+    <div fxLayout="row" fxLayout.lt-md="column">
+
+      <div fxFlex="50%">
+
+        <mat-nav-list>
+
+          <mat-list-item [routerLink]="['']">
```
Comment (abhaychawla) [inline src/app/products/products.component.html]: Remove these blank routerLinks from the list items. They are not required right now.

File: src/app/products/products.component.html [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            <mat-icon matListIcon>
+              <fa-icon icon="tag" size="sm"></fa-icon>
```
Comment (abhaychawla) [inline src/app/products/products.component.html]: Use briefcase icon.

File: src/app/products/products.component.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            </mat-icon>
+            <h4 matLine>Loan Products</h4>
+            <p matLine>Add new loan product or modify or inactivate loan product</p>
+          </mat-list-item>
+
+          <mat-list-item [routerLink]="['']">
+            <mat-icon matListIcon>
+              <fa-icon icon="tag" size="sm"></fa-icon>
```
Comment (abhaychawla) [inline src/app/products/products.component.html]: Use briefcase icon.

File: src/app/products/products.component.html [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            </mat-icon>
+            <h4 matLine>Savings Products</h4>
+            <p matLine>Add new savings product or modify or inactivate savings product</p>
+          </mat-list-item>
+
+          <mat-list-item [routerLink]="['']">
+            <mat-icon matListIcon>
+              <fa-icon icon="tag" size="sm"></fa-icon>
```
Comment (abhaychawla) [inline src/app/products/products.component.html]: Use briefcase icon.

File: src/app/products/products.component.html [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            <mat-icon matListIcon>
+              <fa-icon icon="briefcase" size="sm"></fa-icon>
+            </mat-icon>
+            <h4 matLine>Recurring Deposit Products</h4>
+            <p matLine>Add, modify or inactivate a Recurring Deposit product</p>
+          </mat-list-item>
+
+          <mat-list-item [routerLink]="['']">
+            <mat-icon matListIcon>
+              <fa-icon icon="briefcase" size="sm"></fa-icon>
```
Comment (abhaychawla) [inline src/app/products/products.component.html]: Use cogs icon.

File: src/app/products/products.component.html [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+      </div>
+
+      <div fxFlex="50%">
+
+        <mat-nav-list>
+
+          <mat-list-item [routerLink]="['']">
+            <mat-icon matListIcon>
+              <fa-icon icon="tags" size="sm"></fa-icon>
```
Comment (abhaychawla) [inline src/app/products/products.component.html]: Use random icon.

**Comment by jonathanzhang53** [comment]
@abhaychawla I will make the changes within the hour. Is it possible for you to stay online and approve once I finish?

**Comment by jonathanzhang53** [comment]
@abhaychawla corrected the faicons.

**Comment by abhaychawla** [comment]
Please update the gif and submit the task on the gci portal.

**Comment by abhaychawla** [comment]
Good work, Thanks!

**Pull request merged**